src/app/sportovni-akce/[category]/[id]/page.tsx

```diff
@@ -3,35 +3,62 @@ import Image from 'next/image';
 import Link from 'next/link';
 import { notFound } from 'next/navigation';
 
-import { EventDetail } from './EventDetail';
+import { EventDetail } from '../../../../components/EventDetail';
 import { OwnerCard } from './OwnerCard';
-import { Events } from '../../../components/Events';
-import { getAllEvents } from '../../../api/events';
-import { getEvent } from '../../../api/events';
-import { StarRating } from '../../../components/StarRating';
+import { Events } from '../../../../components/Events';
+import { getAllEvents } from '../../../../api/events';
+import { getEvent } from '../../../../api/events';
+import { StarRating } from '../../../../components/StarRating';
 import { mockEventOwner } from './mock';
-import { defaultEvent } from '../../../utils/constants';
-import { getImagePath } from '../../../utils/functions';
+import {
+  Routes,
+  defaultEvent,
+  levelLabels,
+  sportsLabels,
+} from '../../../../utils/constants';
+import { getImagePath } from '../../../../utils/functions';
+import { formatDate, formatTime } from '../../../../utils/dateUtils';
+import { EmailForm } from '../../../../components/EmailForm';
+import { slugifyCategory } from '../../../../utils/slugifyUtils';
 
 type EventPageProps = {
-  params: { id: string };
+  params: {
+    category: string;
+    id: string;
+  };
 };
 
 const EventPage: NextPage<EventPageProps> = async ({ params }) => {
   const events = (await getAllEvents()) || [];
 
+  const { category, id } = params;
+
   const event =
-    (await getEvent(params?.id)?.catch(() => {
+    (await getEvent(id)?.catch(() => {
       notFound();
     })) || defaultEvent;
```

Comment:
Why is there defaulting to events? If none is found, do optional chaining in that place in the render. Or is it due to some other place?